dali/operators/generic/constant.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/generic/constant.h"
+
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/core/static_switch.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(Constant)
+  .DocStr(R"code(Produces a batch of constant tensors.
+
+The floating point input data should be placed in `fdata` argument and integer data in `idata`.
+The data is a flat vector of values or a single scalar. The data is then reshaped according
+to the `shape` argument. If the data is scalar, it will be broadcast to fill the entire shape.
+
+The operator only performs meaningful work at first invocation - subsequent calls will return
+a reference to the same memory.
+
+The operator can be automatically instantiated in Python with a call to
+`dali.types.Constant(value, dtype, shape, layout)`. The value can be a scalar, a tuple, a list or
+a numpy array, in which case the `shape` and `dtype` (if not explicitly overridden), will be
+taken from that array.
+
+64-bit integer and double precision arrays are not supported and will be silently downgraded
+to 32-bit.
+)code")
+  .NumInput(0)
+  .NumOutput(1)
+  .AddOptionalArg<int>("shape", "The desired shape of the output. "
+                                 "If not set, the data is assumed to be 1D",
+                  std::vector<int>())
+  .AddOptionalArg<float>("fdata", "Contents of the constant produced (for floating point types).",
```

Comment:
Shouldn't it be prohibited to pass both of `fdata` and `idata` to one node. I think we should provide an error in such case.

Reply:
We do?

Reply:
Can you also warn the user in the doc string?